libgnucash/engine/ScrubBudget.c

```diff
@@ -0,0 +1,238 @@
+/********************************************************************\
+ * ScrubBudget.c -- fix budget amount signs                         *
+ *                                                                  *
+ * This program is free software; you can redistribute it and/or    *
+ * modify it under the terms of the GNU General Public License as   *
+ * published by the Free Software Foundation; either version 2 of   *
+ * the License, or (at your option) any later version.              *
+ *                                                                  *
+ * This program is distributed in the hope that it will be useful,  *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of   *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the    *
+ * GNU General Public License for more details.                     *
+ *                                                                  *
+ * You should have received a copy of the GNU General Public License*
+ * along with this program; if not, contact:                        *
+ *                                                                  *
+ * Free Software Foundation           Voice:  +1-617-542-5942       *
+ * 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652       *
+ * Boston, MA  02110-1301,  USA       gnu@gnu.org                   *
+ *                                                                  *
+\********************************************************************/
+
+#include <config.h>
+
+#include <glib.h>
+#include <glib/gi18n.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdint.h>
+
+#include "gnc-budget.h"
+#include "gnc-features.h"
+#include "ScrubBudget.h"
+
+static QofLogModule log_module = "gnc.engine.scrub";
+
+typedef enum
+{
+    HEURISTICS_INC_EXP,
+    HEURISTICS_CREDIT_ACC,
+    HEURISTICS_NONE
+} HeuristicsOutput;
+
+typedef struct
+{
+    GncBudget* budget;
+    HeuristicsOutput policy;
+} Reversal_t;
+
+typedef struct
+{
+    gint asset,liability,equity,income,expense;
+    gint num_periods;
+    GncBudget* budget;
+} Heuristics;
```

Comment:
These aren't heuristics, they're accumulators (most of which are used only to print a progress message) and a couple of arguments for `heuristics_on_budget` to pass to `process_heuristics_acct`.  Something like `ProcessData` would be more descriptive.